code/parachain/frame/assets-transactor-router/src/lib.rs

```diff
@@ -0,0 +1,918 @@
+			#[pallet::compact] amount: T::Balance,
+		) -> DispatchResult {
+			Pallet::<T>::ensure_admin_or_governance(origin, &asset_id)?;
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::mint_into(asset_id, &dest, amount)?;
+			Ok(())
+		}
+
+		/// Burns `amount` of `asset_id` into the `dest` account.
+		#[pallet::weight(T::WeightInfo::burn_from())]
+		pub fn burn_from(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			dest: <T::Lookup as StaticLookup>::Source,
+			#[pallet::compact] amount: T::Balance,
+		) -> DispatchResult {
+			Pallet::<T>::ensure_admin_or_governance(origin, &asset_id)?;
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungibles::Mutate<T::AccountId>>::burn_from(asset_id, &dest, amount)?;
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Returns `Ok(())` if origin is root or asset is signed by root or by origin
+		pub(crate) fn ensure_admin_or_governance(
+			origin: OriginFor<T>,
+			asset_id: &T::AssetId,
+		) -> Result<(), DispatchError> {
+			// TODO: that must be ensure_asset_origin(origin, asset_id))
+			if T::AdminOrigin::ensure_origin(origin.clone()).is_ok() {
+				return Ok(())
+			}
+
+			match origin.into() {
+				Ok(frame_system::RawOrigin::Signed(account)) => {
+					match T::GovernanceRegistry::get(asset_id) {
+						Ok(SignedRawOrigin::Root) => Ok(()), /* ISSUE: it says if */
+						// (call_origin.is_signed &&
+						// asst_owner.is_root) then allow
+						// mint/burn -> anybody can mint and
+						// burn PICA?
+						// TODO: https://app.clickup.com/t/37h4edu
+						Ok(SignedRawOrigin::Signed(acc)) if acc == account => Ok(()),
+						_ => Err(DispatchError::BadOrigin),
+					}
+				},
+				Ok(frame_system::RawOrigin::Root) => Ok(()),
+				_ => Err(DispatchError::BadOrigin), /* ISSUE: likely will not support collective
+													* origin which is reasonable to have for
+													* governance
+													https://app.clickup.com/t/37h4edu
+													*/
+			}
+		}
+	}
+
+	impl<T: Config> CreateAsset for Pallet<T> {
+		type LocalAssetId = T::AssetId;
+		type ForeignAssetId = T::AssetLocation;
+
+		fn create_local_asset(
+			protocol_id: [u8; 8],
+			nonce: u64,
+			name: Vec<u8>,
+			symbol: Vec<u8>,
+			decimals: u8,
+			ratio: Option<Rational64>,
+		) -> Result<Self::LocalAssetId, DispatchError> {
+			let bytes = protocol_id
+				.into_iter()
+				.chain(nonce.to_le_bytes())
+				.collect::<Vec<u8>>()
+				.try_into()
+				.expect("[u8; 8] + bytes(u64) = [u8; 16]");
+			let asset_id = Self::LocalAssetId::from(u128::from_le_bytes(bytes));
+
+			T::AssetLookup::register_asset(asset_id, None, ratio, name, symbol, decimals)?;
+
+			Ok(asset_id)
+		}
+
+		fn create_foreign_asset(
+			foreign_asset_id: Self::ForeignAssetId,
+			name: Vec<u8>,
+			symbol: Vec<u8>,
+			decimals: u8,
+			ratio: Option<Rational64>,
+		) -> Result<Self::LocalAssetId, DispatchError> {
+			let asset_id = Self::LocalAssetId::from(u128::from_be_bytes(sp_core::blake2_128(
```

Comment:
Good to have a comment on the implementation of unique id how it's supposed to work with multiasset/multilocation. Specially since this looks non-unique without detail